compressors/integer_coders.py

```diff
@@ -85,3 +86,149 @@ def test_universal_uint_encode_decode():
 
     is_lossless, _ = try_lossless_compression(data_block, encoder, decoder)
     assert is_lossless
+
+
+"""
+shared initialization for Golomb encoder and decoder
+"""
+
+
+def _initialize_golomb_coder(coder, M: int):
```

Comment:
I think this is a good time to think about how we want to handle common params for encoders, decoders. 
1. One option is the way you have.. have a function which takes in a `coder` and initializes the params
2. Second option is to explicitly have an object called:
```
class GolombCodeParams:
    def __init__(self, M: ):
        self.M = M 
        self.b = ... 
        self.cutoff = ...
        self.is_rice_code = ...
 
 
 class GolombUintEncoder(DataEncoder):
     def __init__(self, M):
         self.params = GolombParams(M)
```

3. The third option is to have GolombUintEncoder subclass both (DataEncoder and GolombParams)
     Not very familiar with multiple inheritance, but probably should work

Reply:
Do options 2 and 3 have any advantage over option 1? We can discuss more when we talk.

Reply:
Just want to make sure we don't add complications without any reason.

Reply:
yes, I agree we should definitely think if either of those options adds complications. 
(as this scenario of encoder/decoder sharing params is not uncommon) 

I like the option 2, as it makes things a bit more explicit. We could also use `@dataclass` to simplify a few things. 
Option 1, we are modifying an object, which felt less elegant to me.  Lets continue discussing